src/inmanta/user_setup.py

```diff
@@ -0,0 +1,181 @@
+            )
+
+        click.echo(f"{'Server authentication method: ' : <50}{click.style('database', fg='green')}")
+
+        # make sure there is auth config that supports signing tokens
+        cfg = config.AuthJWTConfig.get_sign_config()
+        if cfg is None:
+            click.echo("Error: No signing config available in the configuration.")
+            click.echo("To use a new config, add the following to the configuration in /etc/inmanta/inmanta.d/auth.cfg:\n")
+            click.echo(generate_signing_config())
+            raise click.ClickException(
+                "Make sure signing configuration is added to the config. See the documentation for details."
+            )
+
+        click.echo(f"{'Authentication signing config: ' : <50}{click.style('found', fg='green')}")
+
+    else:
+        raise click.ClickException("The user setup was aborted as it was not executed locally on the orchestrator")
+
+
+async def get_connection_pool() -> asyncpg.pool.Pool:
+    """
+    connect to a connection pool. It is the responsibility of the caller to close the return connection pool
+    """
+    try:
+        database_host = server_config.db_host.get()
+        database_port = server_config.db_port.get()
+
+        database_username = server_config.db_username.get()
+        database_password = server_config.db_password.get()
+        connection_pool_min_size = server_config.db_connection_pool_min_size.get()
+        connection_pool_max_size = server_config.db_connection_pool_max_size.get()
+        connection_timeout = server_config.db_connection_timeout.get()
+        return await data.connect(
+            database_host,
+            database_port,
+            server_config.db_name.get(),
+            database_username,
+            database_password,
+            create_db_schema=False,
+            connection_pool_min_size=connection_pool_min_size,
+            connection_pool_max_size=connection_pool_max_size,
+            connection_timeout=connection_timeout,
+        )
+    except Exception:
+        raise ConnectionPoolException
+
+
+async def do_user_setup() -> None:
+    """Perform the user setup that requires the database interaction"""
+    connection = None
+    try:
+        click.echo(f"{'Trying to connect to DB:' : <50}{server_config.db_host.get()}")
+        connection = await get_connection_pool()
+        DBschema = schema.DBSchema(CORE_SCHEMA_NAME, PACKAGE_WITH_UPDATE_FILES, connection)
+        schema_up_to_date = await DBschema.is_db_schema_up_to_date()
+        if schema_up_to_date:
+            click.echo(f"{'DB schema up to date' : <50}{click.style('yes', fg='green')}")
+        else:
+            click.echo(
+                f"{'DB schema up to date' : <50}"
+                f"{click.style('no: please make sure your DB version and software version are aligned', fg='red')}"
+            )
+            raise click.ClickException("The database version and software version are not aligned")
```

Comment:
I think I see what you mean now. It's also important that the version of the client used is the same as the version on the server. I am starting to think that it wasn't a good idea to add this very strong constraint. It will be rather hard for end-user to use this command properly. Users have to make sure that:

* They use the `inmanta-initial-user-setup` command in the venv of the inmanta server or they need to create a venv themselves with the exact same version as the inmanta server.
* And, they need to make sure that the server executed all the database migration scripts to the latest version. 

We will need to do quite some explaining in the documentation of this command and in the error messages to make sure that users can use it without much trouble. @sanderr Are you still in favor of adding this strong constraint?

Reply:
@sanderr can I assume your thumps up on the previous comment means that you are still in favor of adding the constraint if the message in the exception is made clearer?

Reply:
I am but @arnaudsjs is more hesitant and he does make some valid points. We decided we wanted to involve @wouterdb in this. Forgot to notify you, sorry about that.

Reply:
Multiple remarks here:
1. the error message should guide the user to a solution, i.e. it has to tell the user what to do.
2. I think using a venv is unlikely, but not impossible.
3. I would not check the DB version but try the query and see if it sticks. If there are DB failures that are not due to authentication, I would assume the schema is not up to date an prompt the user to start the orchestrator first.